akka-http2-support/src/main/scala/akka/http/impl/engine/http2/ResponseRendering.scala

```diff
@@ -4,13 +4,33 @@
 
 package akka.http.impl.engine.http2
 
-import akka.http.scaladsl.model.{ ContentType, ContentTypes, HttpHeader, HttpResponse }
+import akka.event.LoggingAdapter
+import akka.http.impl.util.StringRendering
+import akka.http.scaladsl.model.headers.{ Date, Server }
+import akka.http.scaladsl.model._
 import akka.http.scaladsl.model.http2.Http2StreamIdHeader
 
+import scala.annotation.tailrec
 import scala.collection.immutable.VectorBuilder
+import scala.collection.immutable
 
 private[http2] object ResponseRendering {
-  def renderResponse(response: HttpResponse): Http2SubStream = {
+
+  @volatile
+  private var cachedDateHeader = (0L, ("", ""))
+
+  private def dateHeader(): (String, String) = {
+    val cachedSeconds = cachedDateHeader._1
+    val now = System.currentTimeMillis()
+    if (now / 1000 > cachedSeconds) {
+      val r = new StringRendering
+      DateTime(now).renderRfc1123DateTimeString(r)
+      cachedDateHeader = (now, Date.lowercaseName → r.get)
+    }
+    cachedDateHeader._2
+  }
+
+  def renderResponse(serverHeader: Option[Server], log: LoggingAdapter)(response: HttpResponse): Http2SubStream = {
```

Comment:
We could cache the Server name->value tuple instead to save another tuple alloc on each response